wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/Proto2WireProtocCompatibilityTests.kt

```diff
@@ -156,6 +161,40 @@ class Proto2WireProtocCompatibilityTests {
     assertThat(mapTypeWire.map_string_string[""]).isEqualTo("ed")
   }
 
+  /**
+   * The TS-proto library encodes enums differently from Wire or Protoc. Protoc is fine with this,
+   * but it causes Wire to crash. The fix is to make Wire accept this format also.
+   *
+   * https://github.com/stephenh/ts-proto
+   */
+  @Test fun repeatedEnumIsPackedAndLengthIsZero() {
+    val wireMessage = QuiltContainer(
+      quilt = Quilt(
+        fringe = listOf(QuiltColor.GREEN),
+        cozy = true,
+      )
+    )
+
+    // A sample value encoded by ts-proto.
+    val encodedByTsProto = "12091201041a0022003801".decodeHex()
+
+    val googleMessage = RepeatedEnum.QuiltContainer.newBuilder()
+      .setQuilt(
+        RepeatedEnum.Quilt.newBuilder()
+          .addAllFringe(listOf(RepeatedEnum.QuiltColor.GREEN))
+          .setCozy(true)
+          .build()
+      ).build()
+
+    val wireMessageDecodedFromGoogleMessage =
+      QuiltContainer.ADAPTER.decode(encodedByTsProto)
```

Comment:
That's interesting. This line fails on CI with
```
java.io.IOException: Expected to end at 7 but was 8
        at com.squareup.wire.ProtoReader.afterPackableScalar(ProtoReader.kt:385)
        at com.squareup.wire.ProtoReader.readVarint32(ProtoReader.kt:277)
        at com.squareup.wire.EnumAdapter.decode(EnumAdapter.kt:72)
        at com.squareup.wire.EnumAdapter.decode(EnumAdapter.kt:25)
        at squareup.proto2.kotlin.interop.Quilt$Companion$ADAPTER$1.decode(Quilt.kt:205)
        at squareup.proto2.kotlin.interop.Quilt$Companion$ADAPTER$1.decode(Quilt.kt:159)
        at squareup.proto2.kotlin.interop.QuiltContainer$Companion$ADAPTER$1.decode(QuiltContainer.kt:110)
        at squareup.proto2.kotlin.interop.QuiltContainer$Companion$ADAPTER$1.decode(QuiltContainer.kt:82)
```

But protogram which is still on Wire 3.1.0 can deserialize it https://mattprecious.github.io/protogram/?12091201041a0022003801

![image](https://user-images.githubusercontent.com/1767669/236482123-264cdad4-7e09-48f3-88ac-33e56d5e78f2.png)

Did we break in the mean time?